Add custom URL resolution scheme option

gabrielgiordan opened the pull request:
In our scenario, we cannot support `X-Forwarded-Proto` header due to network and security constrains related to our business case, therefore, we'd like to have the option to use a custom `https` scheme for callbacks instead

I think this is also related to https://github.com/ueberauth/ueberauth/issues/103 and https://github.com/ueberauth/ueberauth/issues/65

So, for now I'll just add a custom scheme option, but if and when this PR is approved and merged, I'd like to also add a custom port option, to better solve the issue aforementioned

yordis reviewed (changes requested):
Hey there, would you mind adding an entry in the CHANGELOG and bump the version correctly?

gabrielgiordan commented:
> Hey there, would you mind adding an entry in the CHANGELOG and bump the version correctly?

Sure, I just bumped the minor version and added to CHANGELOG, also added some docs for using it
